api/tests/unit/mappers/test_unit_mappers_engine.py

```diff
@@ -0,0 +1,463 @@
+            id=environment.project.organisation.id,
+            name="Test Org",
+            feature_analytics=False,
+            stop_serving_flags=False,
+            persist_trait_data=True,
+        ),
+        hide_disabled_flags=False,
+        segments=[
+            SegmentModel(
+                id=segment.id,
+                name=segment.name,
+                rules=[],
+                feature_states=[expected_segment_feature_state_model],
+            ),
+        ],
+        enable_realtime_updates=False,
+        server_key_only_feature_ids=[],
+    )
+
+    expected_result = EnvironmentModel(
+        id=environment.id,
+        api_key=environment.api_key,
+        project=expected_project_model,
+        feature_states=[expected_feature_state_model],
+        name=environment.name,
+        allow_client_traits=environment.allow_client_traits,
+        updated_at=environment.updated_at,
+        use_mv_v2_evaluation=environment.use_mv_v2_evaluation,
+        hide_sensitive_data=environment.hide_sensitive_data,
+        amplitude_config=None,
+        segment_config=None,
+        mixpanel_config=None,
+        heap_config=None,
+        dynatrace_config=None,
+        webhook_config=None,
+        hide_disabled_flags=environment.hide_disabled_flags,
+    )
+
+    # When
+    result = engine.map_environment_to_engine(environment)
+    segment_feature_state_uuids = [
+        fs.featurestate_uuid for fs in result.project.segments[0].feature_states
+    ]
+
+    # Then
+    assert (
+        result.project.segments[0].feature_states
+        == expected_result.project.segments[0].feature_states
+    )
+    assert result == expected_result
+    assert (
+        different_environment_segment_feature_state.uuid
+        not in segment_feature_state_uuids
+    )
+    assert not_live_segment_feature_state.uuid not in segment_feature_state_uuids
+    assert segment_featurestate.uuid not in segment_feature_state_uuids
+
+
+def test_map_environment_api_key_to_engine__return_expected(
+    environment: Environment,
+    environment_api_key: "EnvironmentAPIKey",
+) -> None:
+    # Given
+    client_api_key = environment.api_key
+
+    # When
+    result = engine.map_environment_api_key_to_engine(environment_api_key)
+
+    # Then
+    assert result == EnvironmentAPIKeyModel(
+        id=environment_api_key.pk,
+        key=environment_api_key.key,
+        created_at=environment_api_key.created_at,
+        name=environment_api_key.name,
+        client_api_key=client_api_key,
+        expires_at=environment_api_key.expires_at,
+        active=environment_api_key.active,
+    )
+
+
+def test_map_identity_to_engine__return_expected(
+    environment: Environment,
+    identity: "Identity",
+    feature: "Feature",
+    trait: "Trait",
+    identity_featurestate: FeatureState,
+    mocker: MockerFixture,
+) -> None:
+    # Given
+    environment_api_key = environment.api_key
+    expected_result = IdentityModel.construct(
```

Comment:
what's the difference between `.construct()` and just calling the default `__init__()` method here?

Reply:
Maybe worth a comment?